common/FormattingUtils.js

```diff
@@ -31,10 +31,56 @@ export function formatWithMaxDecimals(num, decimalPlaces, roundUp) {
   const fullPrecisionFloat = BigNumber(num);
 
   // Convert back to BN to truncate any trailing 0s that the toFixed() output would print.
-  const fixedPrecisionFloat = BigNumber(fullPrecisionFloat.toFixed(decimalPlaces));
-  return fixedPrecisionFloat.toString();
-}
+  const fixedPrecisionFloat = BigNumber(fullPrecisionFloat).toFixed(decimalPlaces);
+  return fixedPrecisionFloat.toString().replace(/\B(?=(\d{3})+(?!\d))/g, ",");
+};
 
-export function createFormatFunction(web3, numDisplayedDecimals) {
+const createFormatFunction = (web3, numDisplayedDecimals) => {
   return valInWei => formatWithMaxDecimals(formatWei(valInWei, web3), numDisplayedDecimals, false);
+};
+
+// generate a etherscan link prefix
+function createEtherscanLinkFromtx(web3, networkUtils) {
+  // Construct etherscan link based on network
+  // const networkId = web3.networkId;
+  const networkId = 1;
+  let url;
+  if (networkUtils[networkId]) {
+    url = `${networkUtils[networkId].etherscan}`;
+  } else {
+    // No URL for localhost, just show transaction ID
+    url = "";
+  }
+  return url;
+}
+
+// Convert either an address or transaction to a shorter version.
+// 0x772871a444c6e4e9903d8533a5a13101b74037158123e6709470f0afbf6e7d94 -> 0x7787...7d94
+function createShortHexString(hex) {
+  return hex.substring(0, 5) + "..." + hex.substring(hex.length - 6, hex.length - 1);
 }
+
+// Take in either a transaction or an account and generate an etherscan link for the corresponding
+// network formatted in markdown.
+function createEtherscanLinkMarkdown(web3, networkUtils, hex) {
+  let shortURLString = createShortHexString(hex);
+
+  // Transaction hash
+  if (hex.length == 66) {
+    return `<${createEtherscanLinkFromtx(web3, networkUtils)}tx/${hex}|${shortURLString}>`;
+  }
+  // Account
+  else if (hex.length == 42) {
+    return `<${createEtherscanLinkFromtx(web3, networkUtils)}address/${hex}|${shortURLString}>`;
+  }
+}
+
+module.exports = {
```

Comment:
Any reason why you changed the exports to `es6` style (`const func = (args) => {...}` vs `export const....`)?

Reply:
did this to be consistent with the rest of our infrastructure. my one worry is that the voter dapp is impacted by these changes in how it imports these files. Because we've frozen it it's fine for now but will need to double check that changing how these are imported doesn't effect anything upstream.

Reply:
As long as it builds successfully in ci (we still perform the build and test steps for the dapps), I think we should be good.

To double check, though, you may want to just fire it up with `npm start`.

Reply:
Tested in both dapps and they are both able to start correctly with `npm start`. did not validate functionality though but I think we are good.

Reply:
All that matters is that the imports process correctly.